### ISSUE TITLE
test_defer calls reactor.run()

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2052]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2052|
|Type|defect|
|Created|2006-09-02 19:40:35Z|

<details><summary>Searchable metadata</summary>

```
trac-id__2052 2052
type__defect defect
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1157226035000000 1157226035000000
changetime__1157593734000000 1157593734000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun removed owner|
|-|-|

Removed the call, as well as added suppressions for various deprecated behaviors which are explicitly being tested.

Please review in test_defer-cleanups-2052

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set owner to @jml<br>@jml set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_3"></a>|@jml set owner to @exarkun<br>@jml set status to `new`|
|-|-|

Looks fine to me. Well done on separating test_indexingFirstError. Can't really find a negative thing to say -- merge away.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [18070]) Merge test_defer-cleanups-2052

Author: exarkun
Reviewer: jml
Fixes [#2052](https://github.com/twisted/twisted/issues/2052)

This adds several warning suppressions to tests of deprecated Deferred behavior.
It also splits up the FirstError test into a couple pieces, since several different
pieces of functionality were being tested by one method.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
